nexus_constructor/add_component_window.py

```diff
@@ -0,0 +1,258 @@
+from enum import Enum
+
+from PySide2.QtCore import QUrl, Signal, QObject
+from nexus_constructor.qml_models.geometry_models import (
+    CylinderModel,
+    OFFModel,
+    NoShapeModel,
+)
+from PySide2.QtGui import QValidator
+from ui.add_component import Ui_AddComponentDialog
+from nexus_constructor.component_type import (
+    make_dictionary_of_class_definitions,
+    PIXEL_COMPONENT_TYPES,
+)
+from nexus_constructor.validators import UnitValidator, NameValidator
+from nexus_constructor.nexus_wrapper import NexusWrapper
+from nexus_constructor.utils import file_dialog, validate_line_edit
+import os
+from functools import partial
+
+GEOMETRY_FILE_TYPES = {"OFF Files": ["off", "OFF"], "STL Files": ["stl", "STL"]}
+
+
+class FileValidator(QValidator):
```

Comment:
I suggest naming it more specifically `GeometryFileValidator` and moving it to its own file.